src/page-modules/assistant/trip/index.tsx

```diff
@@ -0,0 +1,177 @@
+import { formatLocaleTime, secondsToDuration } from '@atb/utils/date';
+import {
+  TripData,
+  type TripPattern,
+} from '@atb/page-modules/assistant/server/journey-planner/validators';
+import style from './trip.module.css';
+import { PageText, useTranslation } from '@atb/translations';
+import { Typo } from '@atb/components/typography';
+import { motion } from 'framer-motion';
+import { MonoIcon } from '@atb/components/icon';
+import { TransportIconWithLabel } from '@atb/components/transport-mode/transport-icon';
+import { GeocoderFeature } from '@atb/page-modules/departures';
+import { TransportModeFilterOption } from '@atb/components/transport-mode-filter/types';
+import { nextTripPatterns } from '@atb/page-modules/assistant/client';
+import { createTripQuery } from '@atb/page-modules/assistant/utils';
+import { useEffect, useState } from 'react';
+import { getInitialTransportModeFilter } from '@atb/components/transport-mode-filter/utils';
+import { Button } from '@atb/components/button';
+
+export type TripProps = {
+  initialFromFeature: GeocoderFeature;
+  initialToFeature: GeocoderFeature;
+  initialTransportModesFilter: TransportModeFilterOption[] | null;
+  trip: TripData;
+  departureType: 'arrival' | 'departure';
+};
+
+function tripPatternsWithTransitionDelay(tripPatterns: TripPattern[]) {
+  return tripPatterns.map((tripPattern, i) => ({
+    ...tripPattern,
+    transitionDelay: i * 0.1,
+  }));
+}
+
+export function Trip({
+  initialFromFeature,
+  initialToFeature,
+  initialTransportModesFilter,
+  trip,
+  departureType,
+}: TripProps) {
+  const { t } = useTranslation();
+  const [tripPatterns, setTripPatterns] = useState(
+    tripPatternsWithTransitionDelay(trip.tripPatterns),
+  );
+  const [nextPageCursor, setNextPageCursor] = useState(trip.nextPageCursor);
+  const [previousPageCursor, setPreviousPageCursor] = useState(
+    trip.previousPageCursor,
+  );
+  const [isFetchingTripPatterns, setIsFetchingTripPatterns] = useState(false);
+
+  useEffect(() => {
+    setTripPatterns(tripPatternsWithTransitionDelay(trip.tripPatterns));
+    setNextPageCursor(trip.nextPageCursor);
+    setPreviousPageCursor(trip.previousPageCursor);
+  }, [trip]);
+
+  const fetchNextTripPatterns = async () => {
+    setIsFetchingTripPatterns(true);
+    const tripQuery = createTripQuery(
+      initialFromFeature,
+      initialToFeature,
+      getInitialTransportModeFilter(initialTransportModesFilter) || undefined,
+    );
+    const cursor =
+      departureType === 'arrival' ? previousPageCursor : nextPageCursor;
+    const trip = await nextTripPatterns(tripQuery, cursor);
+    setTripPatterns([
+      ...tripPatterns,
+      ...tripPatternsWithTransitionDelay(trip.tripPatterns),
+    ]);
+    departureType === 'arrival'
+      ? setPreviousPageCursor(trip.previousPageCursor)
+      : setNextPageCursor(trip.nextPageCursor);
+
+    setIsFetchingTripPatterns(false);
+  };
```

Comment:
I think this can be extracted as a custom hook to clean up the function here and make it more easy to read.

Reply:
Updated this with a custom hook now! 👍